src/test/regress/sql/local_table_join.sql

```diff
@@ -336,6 +336,17 @@ ALTER TABLE local DROP column key1;
 SELECT COUNT(*) FROM distributed_table JOIN local ON distributed_table.value = 'text';
 
 
+-- issue 4682
+create table tbl (a int, b int, c int, d int);
```

Comment:
Is it possible that this query returns some rows? Maybe introduce custom tables instead of `pg_type ` and compare the results with all local tables?